test-suite/shared/src/test/require-jdk17/org/scalajs/testsuite/javalib/util/random/RandomGeneratorTest.scala

```diff
@@ -0,0 +1,433 @@
+    assertEquals(-1591520875, make(0xa12351957679f21fL).nextInt())
+    assertEquals(-535590572, make(0xe01389542264f955L).nextInt())
+    assertEquals(323545793, make(0x1348eac1e0b1d30dL).nextInt())
+    assertEquals(1231429688, make(0x496620384f075af6L).nextInt())
+    assertEquals(-945352075, make(0xc7a71275b30dde50L).nextInt())
+    assertEquals(-10741059, make(0xff5c1abdc835a88dL).nextInt())
+    assertEquals(2094467582, make(0x7cd709fed6fc0e33L).nextInt())
+    assertEquals(1165941437, make(0x457edabdccfc99b3L).nextInt())
+    assertEquals(-2038494290, make(0x867f0bae045cc0b5L).nextInt())
+    assertEquals(-653194468, make(0xd9110b1c6a647a80L).nextInt())
+    assertEquals(670614366, make(0x27f8c35e876aa500L).nextInt())
+    assertEquals(1879332287, make(0x700455bf2802ddacL).nextInt())
+    assertEquals(112541956, make(0x06b54104f821ddfeL).nextInt())
+    assertEquals(-1086617849, make(0xbf3b87070bbc6313L).nextInt())
+    assertEquals(1962565644, make(0x74fa600cd5741c70L).nextInt())
+  }
+
+  @Test def testNextIntWithBound(): Unit = {
+    assertEquals(353, make(0xcdff356ba8f073c7L).nextInt(1234))
+    assertEquals(652863253, make(0x4dd3ce2b9e4f98d3L).nextInt(987654321))
+    assertEquals(1546425308, make(0xb85927b9d6e3249bL).nextInt(2000000000))
+    assertEquals(6, make(0x0bf3df8c2f645e7dL).nextInt(10))
+    assertEquals(0, make(0x857f63be5f2f9ca0L).nextInt(1))
+    assertEquals(1351723210, make(0xa12351957679f21fL).nextInt(Int.MaxValue))
+
+    // Rejects one sample
+    assertEquals(1047233791, make(0xff5c1abdc835a88dL, 0x7cd709fed6fc0e33L).nextInt(2000000000))
+
+    // Powers of 2 have a dedicated path
+    assertEquals(5, make(0x457edabdccfc99b3L).nextInt(8))
+    assertEquals(942, make(0x867f0bae045cc0b5L).nextInt(1024))
+    assertEquals(420547356, make(0xd9110b1c6a647a80L).nextInt(1 << 30))
+
+    assertThrows(classOf[IllegalArgumentException], make().nextInt(0))
+    assertThrows(classOf[IllegalArgumentException], make().nextInt(-10))
+    assertThrows(classOf[IllegalArgumentException], make().nextInt(Int.MinValue))
+  }
+
+  @Test def testNextIntWithOriginAndBound(): Unit = {
+    assertEquals(3353, make(0xcdff356ba8f073c7L).nextInt(3000, 4234))
+    assertEquals(529406464, make(0x4dd3ce2b9e4f98d3L).nextInt(-123456789, 987654321))
+    assertEquals(-453574692, make(0xb85927b9d6e3249bL).nextInt(-2000000000, -123456789))
+    assertEquals(-4, make(0x0bf3df8c2f645e7dL).nextInt(-10, 0))
+
+    assertEquals(56, make(0x857f63be5f2f9ca0L).nextInt(56, 57)) // no choice
+
+    // Close to min or max
+    assertEquals(-2147483647, make(0xa12351957679f21fL).nextInt(Int.MinValue, Int.MinValue + 3))
+    assertEquals(2147483645, make(0xe01389542264f955L).nextInt(Int.MaxValue - 3, Int.MaxValue))
+    assertEquals(161772897, make(0x1348eac1e0b1d30dL).nextInt(1, Int.MaxValue))
+
+    // Rejects one sample
+    assertEquals(47233791, make(0xff5c1abdc835a88dL, 0x7cd709fed6fc0e33L).nextInt(-1000000000, 1000000000))
```

Comment:
Here it might be warranted to not share RandomGenerator to avoid very difficult to debug test failures.